docs/src/pages/components/speed-dial/OpenIconSpeedDial.js

```diff
@@ -62,20 +57,16 @@ export default function OpenIconSpeedDial() {
         className={classes.speedDial}
         hidden={hidden}
         icon={<SpeedDialIcon openIcon={<EditIcon />} />}
-        onBlur={handleClose}
-        onClick={handleClick}
         onClose={handleClose}
-        onFocus={handleOpen}
-        onMouseEnter={handleOpen}
-        onMouseLeave={handleClose}
+        onOpen={handleOpen}
         open={open}
       >
         {actions.map(action => (
           <SpeedDialAction
             key={action.name}
             icon={action.icon}
             tooltipTitle={action.name}
-            onClick={handleClick}
+            onClick={handleClose}
```

Comment:
Might it be more helpful to have a click handler that shows how to identify which action has been clicked, before it calls onClose?

Reply:
It could help, it would make the demo a bit more verbose, but could teach people an valuable pattern.

Reply:
From personal experience, it's the kind of thing React beginners struggle with. I think it might be work a little verbosity.

Reply:
Yes, I think that people can leverage it when displaying data from the server, for instance, in the case of inbox (R.I.P.), the most frequent contacts. In the case of Google Contacts, I would expect each action to have a custom handler.

Reply:
I'm not sure how to action this comment. I would expect most people to have a fixed set of options (not using an array). However, our demo is using an array for the sake of minimizing the demo length. I believe these two objectives contradict themselves.

Reply:
Not sure why it matters. In either case calling a common click handler that identifies the action clicked. Some of our demos throw up an alert: "You clicked the <label> action."

Reply:
Sure, I have no objection in changing the handler, my point is that I fail to picture what you have in mind.

Reply:
You're thinking that in real usage each action would be a discrete component calling a different handler?

Reply:
I think the point is that the click handler does not handle each action differently which should be the most common use case. Otherwise why have a speed dial if every action does the same (which only closes at the moment). I would expect that you want to dispatch some action. It would be nice if this demo would handle that case. 

At least this is how I understood @mbrookes

Reply:
Ok thanks, this sounds good 👍

Reply:
@eps1lon Yeah, that's the crux of it. I took it from @oliviertassinari's comment that in real usage you wouldn't use a mapped array, but would compose each action component. If each then calls a different click handler, the issue goes away.

I had pictured a common handler that dispatches a different event according to the calling action.

Reply:
Just use an inline handler that captures the action. A mapped array is the first iteration you would do. The rest is premature optimization.

Reply:
I'm going to merge, I think that we can apply the handler changes in another pull request :).